### DISCUSSION TITLE
GitHub CLI issue when attempting to clone repo

### DISCUSSION THREAD

**Discussion by joekolassa** [community/community#45497] [Projects and Issues] [answered]
I just got the GitHub CLI working on my linux VM. All of the commands that I do not need seem to be working just fine. However, the main reason I set up the GitHub CLI was in order to clone my repository to the VM and for some reason I am running into trouble with this command.  

When I run it I always get the same message: "To get started with GitHub CLI, please run: gh auth login.  Alternatively, populate the GH_TOKEN environment variable with a GitHub API authentication token."  I will provide a screenshot to confirm this but I am definitely logged in already using a personal access token.  I have been trying my best google-fu to find others with this same issue but no one seems to be reporting this behavior.  Everything I am finding says that once you are logged in you just run the "gh repo clone" command and it works.  I have seen some talk about providing the GH_TOKEN to help with authentication but I haven't been able to find a comprehensive explanation of how this is used.  That may solve my issue though so any info on how that works would be greatly appreciated.

I am somewhat new to this so if I have not provided all of the necessary details please let me know and I will provide whatever I can to help in resolving this.  Thanks!
![GitHubCLIissue](https://user-images.githubusercontent.com/115965784/215008474-a95acd38-a6f8-446a-8382-e23f69f7d1a4.PNG)

**Comment by jge162** [comment]
Hello @joekolassa to resolve this issue set the GH_TOKEN environment variable with the value of your personal access token. You can do this by running the command:

export GH_TOKEN=your_token_here

You can also try running the command 

gh auth login

and enter your token when prompted.

also you can type help to get help too. 

gh help auth

**Reply by joekolassa** [reply]
Click on your profile dropdown in the top right corner in github and go to settings.  Then scroll down and click on developer settings in the menu on the lefthand side of the page.  Then click on personal access tokens dropdown in the lefthand menu.  There you will find both the fine-grained token and the classic token.  You'll have to decide for yourself which one to use.  I used a classic token.

**Reply by joekolassa** [reply]
@tombohub https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

**Comment by joekolassa** [comment]
Hey @jge162
Thank you for reaching out.  If I am not mistaken, the screenshot I provided in my OP shows that I am currently logged in with the token as you suggested.  It shows a green checkmark next to the token and it also shows the scope that I currently have. Those were the permissions I found online that were needed to clone the repository.  I definitely may be mistaken so if I am please let me know where I am going wrong.

I attempted to set the GH_TOKEN environment variable as you stated and it seems to have worked (it at least did not throw an error).  But after doing that I am still receiving the same message when I attempt to clone the repo.  Do I need to use that GH_TOKEN env variable in my clone command somehow?  I looked at the gh help auth command but there wasn't much in there that I found useful. The only things I saw that may help me is the "gh auth setup-git" command and the "gh auth token" command. The first one I guess may be useful but I haven't found anyone else who needed to use that in my research so far.  The second one correctly prints out the token that I believe should be allowing me to clone my repo.

Here is another screenshot showing what I did after reading your comment. Boxed in blue you can see where I ran the clone command after performing the "export GH_TOKEN=My token" command.  After that I looked at the "gh help auth" info.  Then I ran the "gh auth token" command and it shows my token is correct.
![GitHubCLIissue2](https://user-images.githubusercontent.com/115965784/215012302-342cf133-ac99-44ba-afb3-8e2acf5ad427.PNG)

**Comment by jge162** [comment]
hm, It looks like you have correctly set the GH_TOKEN environment variable and also have the correct permissions to clone the repository. 

issue might be with the clone command itself. Can you try running the command "git clone https://github.com/OWNER/REPO.git" (replace OWNER and REPO with the appropriate values) and see if that works? 

if that fails, you can try running "git clone https://<TOKEN>@github.com/OWNER/REPO.git" (replace TOKEN, OWNER and REPO with appropriate values) and see if that works. 

The second option command will clone the repository using the token you have set in the ->GH_TOKEN environment variable.

hope this works for you.

**Comment by joekolassa** [comment]
@jge162 
Before I ever set up any of the GitHub CLI stuff I had tried cloning it using the first method you mentioned.  However, I got a message stating that this authentication method had been removed on August 13, 2021.  So I was pretty sure that would happen again but I still ran the command to confirm that this is the case.

As for your second suggestion I believe that you may have left something out of the command.  The only difference I see between the first and second command is that the second has an @ symbol after https:// but then in parentheses you state that I need to replace TOKEN with the appropriate value.  I think you forgot to add the part of the command where I need to put the token.

![GitHubCLIissue3](https://user-images.githubusercontent.com/115965784/215014763-2b729842-bf26-422b-9092-8b98588d5985.PNG)

**Comment by jge162** [comment, marked as answer]
You are correct, 

git clone https://TOKEN@github.com/OWNER/REPO.git

replace TOKEN, OWNER, and REPO with the appropriate values.

**Comment by joekolassa** [comment]
Well, I'll be damned.  I believe that worked.  It is very frustrating that the GitHub CLI method does not seem to work for me but I guess that is not that big of a deal.  Thank you for the help!

**Reply by jge162** [reply]
no problem.

**Reply by tombohub** [reply]
it doesn't work for me too

**Comment by bartoszandala** [comment]
Hi, I know this is an old thread, but nevertheless:
You are exporting GH_TOKEN for the user bitnami, and then you are trying to clone with sudo, that means as the user root, which does not have token exported. Try to clone without sudo and it should work as expected.
